testcases/OpTestKexec.py

```diff
@@ -27,143 +27,387 @@
 
 '''
 OpTestKexec
----------
+-----------
+Performs following kexec test (Kexec is a Linux feature that
+lets you restart the operating system quickly by loading a
+new kernel into memory and switching to it, avoiding a full
+computer reboot.)
 
-This test is to validate kexec commands
+Test 1: Load and unload kexec kernel
+Test 2: First Load and then boot into kexec kernel
+Test 3: Load and boot into kexec kernel using force option
+Test 4: Single step load and boot into kexec kernel
+Test 5: Single step kexec in loop
+
+There is no guarantee that above tests are will be executed in the
+same order as they are listed if test are executed using class name.
 '''
 
-import os
-import pexpect
 import unittest
 import OpTestLogger
 
 import OpTestConfiguration
 from common.OpTestSystem import OpSystemState
 from common.Exceptions import CommandFailed
-from common.OpTestError import OpTestError
 from common.OpTestUtil import OpTestUtil
 
 log = OpTestLogger.optest_logger_glob.get_logger(__name__)
 
 class OpTestKexec(unittest.TestCase):
+    """kexec test class"""
+
+    def get_distro(self):
+        """
+        Detects the Linux distribution by checking the contents of the
+        '/etc/os-release' file.
+
+        Returns:
+            str: Detected distribution, which can be 'Ubuntu', 'RHEL', 'SLES',
+                 or None if unknown.
+        """
+
+        distro = None
+        ret = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in (ret[0] or ret[1]):
+            distro = "Ubuntu"
+        elif 'Red Hat' in (ret[0] or ret[1]):
+            distro = 'RHEL'
+        elif 'SLES' in (ret[0] or ret[1]):
+            distro = 'SLES'
+
+        return distro
```

Comment:
@sourabhjains Please reuse the existing utils available with OpTestUtils.py 

https://github.com/open-power/op-test/blob/c0924baae2b162358922ccd0e1379207f18307c7/common/OpTestUtil.py#L2225

Reply:
I was not aware of that API. I will update the patch.

Thanks,
Sourabh